nselib/msrpctypes.lua

```diff
@@ -4523,6 +4523,26 @@ function unmarshall_SERVICE_STATUS(data, pos)
 end
 
 
+--- Unmarshalls a null-terminated Unicode string based upon a 32-bit offset (LPTSTR)
+-- @param arguments The data being processed
+-- @param startpos  The current position within the data
+-- @return The new position
+-- @return The string with null removed
+function unmarshall_lptstr(arguments, startpos)
+
+  local endpos = startpos
+  local s = ""
+
+  while s ~= "\0\0" do
+    -- Reads the next character.
+    s = string.sub(arguments, endpos, endpos + 1)
+
+    -- We add 2 because we are dealing with UTF-16 format.
+    endpos = endpos + 2
+  end
+
+  return startpos, string.sub(arguments, startpos, endpos)
```

Comment:
Do you really mean to return `startpos`? Or perhaps `endpos`?

Reply:
I created this function for my `smb-enum-services` script and I would be needing a `startpos` in there and hence I returned startpos. But you are right, the generic unmarshalling function has to return `endpos` and I will fix it soon.